hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -996,9 +994,11 @@ else if ( batchSize > 1 ) {
 //					formnos[l] = -1;
 //					columns.add( colName );
 					cols[i] = colName;
-					aliases.add( selectable.getAlias( dialect, prop.getValue().getTable() ) );
+					final String columnAlias = selectable.getAlias( dialect, prop.getValue().getTable() );
+					if ( prop.isSelectable() && !aliases.contains( columnAlias ) ) {
```

Comment:
I guess you should add something similar for `formulaAliases` in the if-block above